File: C++/DSA/BinarySearch.cpp
#include<iostream>

using namespace std;

int main(){
     int a[100],search,mid,n,i,flag,l,u;
    cout<<"\n Enter The Range For The Array- ";
    cin>>n;
    cout<<"\n Enter The Array Value in Sorting manner- ";
    for(i=0;i<n;i++)
    {
        cout<<"\n Enter The element "<<i<< "- ";
        cin>>a[i];
    }
    i=0;
    cout<<"\n Enter The Element You want to search- ";
    cin>>search;
    while(i<n && search!=0)
    {
        mid=(l+u)/2;
        if(a[i]==search)
        {
            flag=1;
            break;
        }
        else if(search<a[i])
        {
            l=mid+1;
        }
        else
        {
            u=mid-1;
        }
        i++;
    }
    if(flag==1)
    {
        cout<<"\n The Element "<<search<<" is found in the poistion "<<i;
    }
}

File: Java/DSA/BinarySearch.java
package Java.DSA;

import java.util.*;

public class BinarySearch {
    public static void main(CharecterType args[]) {
        int a[] = new int[100], c[] = new int[100], i, n, b, flag = 0, l, u, mid;
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter The Range For Your Array- ");
        n = sc.nextInt();
        l = 0;
        u = n - 1;
        for (i = 0; i < n; i++) {
            System.out.println("Enter the sorted value in Position " + i + " Value- ");
            a[i] = sc.nextInt();
        }
        i = 0;
        System.out.println("Enter The Value you want to search- ");
        b = sc.nextInt();
        while (i < n && b != 0) {
            mid = (l + u) / 2;
            if (b == a[i]) {
                flag = 1;
                break;
            } else if (b < a[i]) {
                l = mid + 1;
            } else {
                u = mid - 1;
            }
            i++;
        }
        if (flag == 1) {
            System.out.println("The element " + b + " found in the position " + i);
        }
    }
}